### ISSUE TITLE
Fail to load extern Proto from another github repo

### ISSUE THREAD

**Issue by BenjaminDeleze** [cyberbotics/webots#5058]
Webots crash with the following world:

```
#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots-cloud-simulation-examples/test-protos-download/7_local_assets/protos/MyProto.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.0316302248353415 0.02843866445440257 0.9990949760863156 1.6779495280745207
  position -0.05753688709726134 2.5600691798403346 9.999992923742619
}
Background {
  skyColor 0.5 0.5 0.8
}

MyProto {
}

```

`MyProto` is a valid Proto and works well if you open this world: https://github.com/cyberbotics/webots-cloud-simulation-examples/blob/main/7_local_assets/worlds/local_assets_example.wbt

**Comment by ad-daniel** [comment]
No crashes for me in development mode
![a](https://user-images.githubusercontent.com/44834743/184083315-a082846c-5b2c-4280-a93d-823b11edaece.png)

**Comment by BenjaminDeleze** [comment]
I think it is a problem with CadShape, can you try with `EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots-cloud-simulation-examples/main/7_local_assets/protos/MyProto.proto"` instead?

If it is the case, #5015 will maybe fix it

**Comment by ad-daniel** [comment]
Yes it crashes, the issue is in WbUrl, it triggers an assert when trying to resolve the obj relative to the PROTO

**Comment by ad-daniel** [comment]
I have an idea why it would fail, the ImageTextures used by CadShape are shallow nodes... We might need to unshallow them a bit and define a parent regardless

**Comment by BenjaminDeleze** [comment]
It seems that it has been fixed by #5015, I can now load remote CadShape correctly
